src/Bicep.Core/Semantics/Namespaces/DefaultNamespaceProvider.cs

```diff
@@ -38,32 +38,31 @@ public DefaultNamespaceProvider(IResourceTypeProviderFactory resourceTypeLoaderF
         }.ToImmutableDictionary();
     }
 
-    public NamespaceType? TryGetNamespace(
+    public ResultWithDiagnostic<NamespaceType> TryGetNamespace(
         ResourceTypesProviderDescriptor descriptor,
         ResourceScope resourceScope,
         IFeatureProvider features,
         BicepSourceFileKind sourceFileKind)
     {
         // If we don't have a types path, we're loading a 'built-in' type
         if (descriptor.TypesBaseUri is null &&
-            builtInNamespaceLookup.TryGetValue(descriptor.Name) is { } getProvider)
+            builtInNamespaceLookup.TryGetValue(descriptor.Name) is { } delegateFn)
         {
-            return getProvider(descriptor, resourceScope, features, sourceFileKind);
+            return new(delegateFn(descriptor, resourceScope, features, sourceFileKind));
```

Comment:
[nit] `getProvider` is more descriptive than `delegateFn` - why the rename?

Reply:
IIRC I was confused on wether the return valye is a function or an data object. I renamed it to reflect its a function, but I agree that getProvider is more descriptive